utils/zerovec/src/ule/slices.rs

```diff
@@ -72,3 +72,39 @@ where
         T::from_byte_slice_unchecked(bytes)
     }
 }
+
+// Safety (based on the safety checklist on the ULE trait):
+//  1. [T; N] does not include any uninitialized or padding bytes.
+//  2. [T; N] is aligned to 1 byte.
+//  3. The impl of validate_byte_slice() returns an error if any byte is not valid (never).
+//  4. The impl of validate_byte_slice() returns an error if there are leftover bytes (never).
+//  5. The other ULE methods use the default impl.
+//  6. [T; N] byte equality is semantic equality
+unsafe impl<T: ULE, const N: usize> ULE for [T; N] {
+    #[inline]
+    fn validate_byte_slice(slice: &[u8]) -> Result<(), ZeroVecError> {
+        if slice.len() % N != 0 {
+            return Err(ZeroVecError::length::<[T; N]>(slice.len()));
+        }
+
+        let len_t = slice.len() / N;
+        for i in 0..N {
+            T::validate_byte_slice(&slice[i * len_t..(i + 1) * len_t])?
+        }
+        Ok(())
+    }
+}
+
+impl<T: ULE, const N: usize> AsULE for [T; N] {
```

Comment:
Suggestion: `T` here need not be `ULE`.  It can be `AsULE`, and then you set `type ULE = [T::ULE, N]`.